pkg/crosscluster/streamclient/pgconn.go

```diff
@@ -62,7 +62,7 @@ func setupPGXConfig(remote url.URL, options *options) (*pgx.ConnConfig, error) {
 	config.DialFunc = dialer.DialContext
 
 	// If the user hasn't given us an application name.
-	if _, ok := config.RuntimeParams["application_name"]; !ok {
+	if _, ok := config.RuntimeParams["application_name"]; !ok && options != nil {
```

Comment:
Not sure if we should return an error here if `application_name` cannot be determined?

Reply:
I don't think it can be `nil` here based on how we use the result of `processOptions`, so this is mostly a safety check. That being said, I think not using `nil` makes more sense if there's no semantic reason to do so. Will update.